### PR TITLE
Upgrade Error Prone 2.19.1 -> 2.20.0

### LINKED ISSUE #686: Version 0.11.1 seems to be incompatible with Error Prone 2.20.0
### Describe the bug

Similar to
- #622

but with
```
java.lang.AssertionError: java.lang.IllegalArgumentException: Cannot combine scanner suppliers with different implementations of 'MissingRefasterAnnotation': tech.picnic.errorprone.bugpatterns.MissingRefasterAnnotation, com.google.errorprone.bugpatterns.MissingRefasterAnnotation
```
instead of `StringCaseLocaleUsage` when upgrading EP from 2.19.1 to 2.20.0.

### Setup

- Operating system: ubuntu
- Java version: Java version: 17.0.7, vendor: Eclipse Adoptium
- Error Prone version: 2.20.0
- Error Prone Support version: 0.11.1

### PR DESCRIPTION
**Pull request by Stephan202**
Suggested commit message:
```
Upgrade Error Prone 2.19.1 -> 2.20.0 (#685)

Summary of key changes:
- The `MissingRefasterAnnotation` check was contributed to Error Prone,
  and so is deleted here (multiple checks with the same name are not
  supported).
- Similarly, Error Prone now supports the `-XepAllSuggestionsAsWarnings`
  flag out of the box. So the `ErrorProneFork` class is deleted, as it
  has currently no further use.

While there, include a tweak to `run-mutation-tests.sh`.

Fixes #686.

See:
- https://github.com/google/error-prone/releases/tag/v2.20.0
- https://github.com/google/error-prone/compare/v2.19.1...v2.20.0
- https://github.com/PicnicSupermarket/error-prone/compare/v2.19.1-picnic-1...v2.20.0-picnic-1
```

@rickie due to the inclusion of `MissingRefasterAnnotation` in Error Prone 2.20.0 we have a conceptual repeat of #622: once again Error Prone Support is not compatible with the latest Error Prone release. So I think we should aim for a new release in the coming days. _Edit:_ actual user impact reported @ #686.

### CODE REVIEW

File: refaster-support/src/main/java/tech/picnic/errorprone/refaster/AnnotatedCompositeCodeTransformer.java [hunk 1]
```diff
@@ -148,8 +148,7 @@ private static <A extends Annotation> Optional<A> getAnnotationValue(
 
   private static SeverityLevel overrideSeverity(SeverityLevel severity, Context context) {
     ErrorProneOptions options = context.get(ErrorProneOptions.class);
-    SeverityLevel minSeverity =
-        ErrorProneFork.isSuggestionsAsWarningsEnabled(options) ? WARNING : SUGGESTION;
+    SeverityLevel minSeverity = options.isSuggestionsAsWarnings() ? WARNING : SUGGESTION;
```
Comment (Stephan202) [inline refaster-support/src/main/java/tech/picnic/errorprone/refaster/AnnotatedCompositeCodeTransformer.java:151]: Pitest flags this code as not covered, but that's not true: it's just that the tests are in another module. `RefasterTest` will fail if this line is modified.

**Comment by Stephan202** [comment]
The quality gate fails due to coverage on this change being below 80%, which is silly. Will make the gate less strict.

**Comment by Stephan202** [comment]
^ Let's make sure to merge with the custom suggested commit message.

**Review by rickie — APPROVED** [review]
Found one missing space 😋.

Upgrade looks really nice 😄.

**Review by werli — APPROVED** [review]
Looks all reasonable. :+1: 

Thanks @Stephan202 :muscle:

**Comment by rickie** [comment]
Rebased, will merge once :green_circle:.

**Comment by rickie** [comment]
Btw, slightly tweaked release notes as the words didn't add a lot of value IMHO :).

**Pull request merged**